guarddog/analyzer/metadata/potentially_compromised_email_domain.py

```diff
@@ -69,31 +35,35 @@ def detect(self, package_info, path: Optional[str] = None, name: Optional[str] =
 
         emails = self.get_email_addresses(package_info)
 
-        if len(emails) == 0:
+        if len(emails) == 0:  # type: ignore # caused by asbtract method
             # No e-mail is set for this package, hence no risk
             return False, "No e-mail found for this package"
 
         latest_project_release = self.get_project_latest_release_date(package_info)
 
         has_issues = False
         messages = []
-        for email in emails:
-            sanitized_email = email.strip().replace(">", "").replace("<", "")
-            email_domain = sanitized_email.split("@")[-1]
-            domain_creation_date, domain_exists = self._get_domain_creation_date(email_domain)
+        for email in emails:  # type: ignore # caused by asbtract method
```

Comment:
i think we should iterate on distinct domains, not emails.
if we have 20 maintainers with gmail.com accounts we would be doing 20 checks instead of just 1